KiCAD 8 migration

drid opened the pull request.

drid committed:
Initial import to KiCAD 8

Signed-off-by: Ilias Daradimos <judgedrid@gmail.com>

mattogodoy commented:
Thank you so much for this! I'm looking forward to the finished PR 😃

mattogodoy commented on .gitignore:
```diff
@@ -2,4 +2,43 @@
 .pioenvs
 .piolibdeps
 .vscode
-.DS_Store
```
Please keep `.DS_Store` here, as it's a very annoying file generated by Mac OS every time you open a directory using finder.

drid committed:
Update to KiCAD9
Create missing symbols
Replace symbols with KiCAD symbols
Link symbols with footprints

Signed-off-by: Ilias Daradimos <judgedrid@gmail.com>

drid commented:
Updated to KiCAD 9.
Schematic is fully migrated using kicad symbols where possible.
Added library for project specific symbols
Schematic passes DRC
Changed footprints only for connectors and ESP32 for now, let me know if you want me to change the rest, there will be visually a bit different.

On the PCB there are several problems.

1. Connectors are to close to the edge, if ther is no mechanical reason for them to be so far out they can be moved and also be replaced with polarized ones (JST,Molex etc)
<img width="403" height="829" alt="image" src="https://github.com/user-attachments/assets/802be08e-1fe3-4841-80d9-0df70d9a258e" />

2. ESP32 board needs a copper free zone around it's antenna to work properly, as you can see there are several components and tracks inside the keep out zone. I can move things around to help with this. 
<img width="928" height="462" alt="image" src="https://github.com/user-attachments/assets/c55cd9d4-6a48-4dca-90f2-f58f006a11c7" />

3. The power module is not specified, is there a part number? I can design one to go on the PCB if you like.

4. What are these circles for?
<img width="655" height="148" alt="image" src="https://github.com/user-attachments/assets/f50fa325-b065-4ff6-85ac-86a9b9d9677e" />

5. Is there a STEP (or even better FreeCAD) file for the case so I can put everything together and check that the do not intersect?

mattogodoy commented:
Hey! Thanks a lot for this :)
I can see that you know way more than me about this topic, so feel free to change anything that you think can be improved.
Let me answer your questions in order:

1. Moving the connections further away from the edge sounds like a good idea. I have not specified any connectors because I solder the wires directly for two reasons: Lack of space inside the enclosure, and I have bad experience with connectors in environments with tons of vibrations like this one.
2. Sounds great to me! This is a newbie mistake I made.
3. The power module is a tiny buck converter like this one (I don't really know the part number, or even if there is one):
<img width="1000" height="750" alt="image" src="https://github.com/user-attachments/assets/89c845e9-9b06-48bb-aa94-0374ac7c26bb" />
If you can design one to go directly on the PCB it would be absolutely awesome.


4. Those circles are only a visual aid to see where the screws that support the case will be placed, so I did not put any components on top of them because there is no space in the enclosure.
5. I really wish there was, but I'm afraid all the information there is available for this enclosure is this: https://conteg.es/en/product/series-32/

Thanks again for your work and contributions! I really appreciate it.

drid commented:
I would suggest to merge this merge request since the migration to kicad is complete and I will open a new one with the proposed changes.

drid commented:
Which enclosure part number is compatible? There are 2 sizes (100/120mm)

mattogodoy commented:
The correct size is `100X75X20`

Here are some schematics I found (I'll add them to the docs when I find some time):
<img width="1422" height="846" alt="0" src="https://github.com/user-attachments/assets/8f2629c1-183d-4ae1-a779-cb60977505c2" />

drid committed:
Move kicad files to kicad folder

Signed-off-by: Ilias Daradimos <judgedrid@gmail.com>
